lib/config.js

```diff
@@ -48,7 +48,12 @@ config.mongodb.connectOptions = {
     journal: true
   },
   server: {
-    auto_reconnect: true
+    socketOptions: {
+      autoReconnect: true, // reconnect on connection failure
```

Comment:
This seems fine. Would be nice to add an autoreconnect test, but it does look like the code will accept either `auto_reconnect` (legacy) or this new fix.